src/map/latent_effect_container.cpp

```diff
@@ -1785,3 +1769,33 @@ void CLatentEffectContainer::CheckLatentsZone(uint16 weather)
     }
     m_POwner->UpdateHealth();
 }
+
+/************************************************************************
+*																		*
+*  Checks all latents regarding current weather							*
+*																		*
+************************************************************************/
+
+void CLatentEffectContainer::CheckLatentsWeather(uint16 weather)
+{
+    for (uint16 i = 0; i < m_LatentEffectList.size(); ++i)
+    {
+        if (m_LatentEffectList.at(i)->GetConditionsID() == LATENT_WEATHER_ELEMENT)
+        {
+            // get the scholar weather and convert to a weather element
+            auto sch_weather = zoneutils::GetWeatherElement(battleutils::GetScholarWeather(m_POwner));
+
+            // if either the new weather or the scholar weather satisfy the latent then activate it
```

Comment:
It just leads me to putting in something like:

WEATHER GetWeather(CBattleEntity* PEntity, bool ignoreScholar, WEATHER zoneWeather = MAX_WEATHER_ID);

So that we can easily determine this call came from either:
- CheckLatentsWeather(...)
- Anywhere else

If that is fine, I will make changes in that direction.